### PR TITLE
E2E setup tests: Ignore Docker build ./wordpress-xdebug

### LINKED ISSUE #4585: E2E - Docker build fails when running inside GH actions
See p1660267127466569-slack-CGGCLBN58

#### Issue 

Sometimes we see the `wordpress` Docker build process failed like [here](https://github.com/Automattic/woocommerce-payments/runs/7800300481?check_suite_focus=true#step:14:35):

```
===> Starting CLIENT containers
Creating network "env_default" with the default driver
Pulling db (mariadb:10.5.8)...
Building wordpress
The command '/bin/sh -c pecl install xdebug-2.9.8 	&& echo 'xdebug.remote_enable=1' >> $PHP_INI_DIR/php.ini 	&& echo 'xdebug.remote_port=9000' >> $PHP_INI_DIR/php.ini 	&& echo 'xdebug.remote_host=host.docker.internal' >> $PHP_INI_DIR/php.ini 	&& echo 'xdebug.remote_autostart=0' >> $PHP_INI_DIR/php.ini 	&& docker-php-ext-enable xdebug > /dev/null' returned a non-zero code: 1
Service 'wordpress' failed to build : Build failed
```

### PR DESCRIPTION
**Pull request by htdat**
Fixes #4585 

See p1660267127466569-slack-CGGCLBN58

#### Issue 

Sometimes we see the `wordpress` Docker build process failed like [here](https://github.com/Automattic/woocommerce-payments/runs/7800300481?check_suite_focus=true#step:14:35):

```
===> Starting CLIENT containers
Creating network "env_default" with the default driver
Pulling db (mariadb:10.5.8)...
Building wordpress
The command '/bin/sh -c pecl install xdebug-2.9.8 	&& echo 'xdebug.remote_enable=1' >> $PHP_INI_DIR/php.ini 	&& echo 'xdebug.remote_port=9000' >> $PHP_INI_DIR/php.ini 	&& echo 'xdebug.remote_host=host.docker.internal' >> $PHP_INI_DIR/php.ini 	&& echo 'xdebug.remote_autostart=0' >> $PHP_INI_DIR/php.ini 	&& docker-php-ext-enable xdebug > /dev/null' returned a non-zero code: 1
Service 'wordpress' failed to build : Build failed
``` 

The purpose of the Dockerfile build `tests/e2e/env/wordpress-xdebug/Dockerfile` is to add xDebug, some other commands like `subversion mariadb-client less`, and WP-CLI. But we're not using them or they have already been available in the WordPress docker image: 

- We disabled xDebug for CI in the `setup.sh` script after building a new image for it: 
https://github.com/Automattic/woocommerce-payments/blob/2db0e16ddb0d5932888d9f8d7760abd6d2fd4063/tests/e2e/env/setup.sh#L108-L112

- We also use `wordpress:cli` in this `setup.sh` script so no need to embed it in the Dockerfile build:
https://github.com/Automattic/woocommerce-payments/blob/2db0e16ddb0d5932888d9f8d7760abd6d2fd4063/tests/e2e/env/setup.sh#L139-L140

#### Changes proposed in this Pull Request
 
My approach here is pretty aggressive, as I am completely removing the Dockerfile file for build. Though, it might be still useful if we need to eable xDebug for local E2E test sites. If so, we might update this PR to have more flexibility between "a build" for local vs "an image" for CI. 

An added benefit is that without building this Dockerfile, the `Prepare test environment` step is faster between 30-60s. 

#### Testing instructions

- Manually run E2E tests for this branch `e2e/ignore-docker-build-for-client-site` on https://github.com/Automattic/woocommerce-payments/actions/workflows/e2e-test.yml
- Confirm that all tests run fine.

Here is a run: https://github.com/Automattic/woocommerce-payments/actions/runs/2845444886

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: tests/e2e/env/docker-compose.yml [hunk 1]
```diff
@@ -2,8 +2,7 @@ version: '3'
 
 services:
   wordpress:
-    build: ./wordpress-xdebug
-    image: wcp_e2e_wordpress
+    image: wordpress:5.6.2-php7.3
```
Comment (achyuthajoy) [inline tests/e2e/env/docker-compose.yml:5]: Could we also update the image to one with PHP v7.4? We specify PHP 7.4 for E2E tests, but the docker images still uses PHP 7.3.

File: tests/e2e/env/setup.sh [hunk 2]
```diff
@@ -105,12 +105,6 @@ if [[ -z $CI ]]; then
 	docker-compose -f "$E2E_ROOT"/env/docker-compose.yml up --build --force-recreate -d phpMyAdmin
 fi
 
-if [[ -n $CI ]]; then
```
Comment (achyuthajoy) [inline tests/e2e/env/setup.sh:108]: @htdat Like you mentioned, on previous modifications, I added this condition to allow xDebug for debugging locally. I'm OK with this change, let see if other have more queries.

**Review by achyuthajoy — COMMENTED** [review]
Thanks for working on this @htdat and I've left some comments. The changes look good to me but haven't approved it. Let hear from other as well.

Also noting that this is the first time I'm seeing the docker build failing. Manual re-run solved the issue and I believe that the failure could be related to GH actions.

File: tests/e2e/env/docker-compose.yml (hunk 1 above)
Comment (htdat) [reply tests/e2e/env/docker-compose.yml:5]: This image ID is from the to-be-deleted `Dockerfile`:
https://github.com/Automattic/woocommerce-payments/blob/5d19129845e132624bb08cf8f7bb325803f04516/tests/e2e/env/wordpress-xdebug/Dockerfile#L1

But yeah, I can add it to PHP 7.4. I will update the WP version to `5.8` too as it's our minimum requirement now:
https://github.com/Automattic/woocommerce-payments/blob/5d19129845e132624bb08cf8f7bb325803f04516/woocommerce-payments.php#L13-L14

Comment (achyuthajoy) [reply tests/e2e/env/docker-compose.yml:5]: That would be great. Thanks 🤝

**Comment by htdat** [comment]
> Also noting that this is the first time I'm seeing the docker build failing. 

This is not my first time though. However, even so, letting Docker build a new image and then not use additional features is kinda... wasteful. 

In the past, we might need these additional features but from what I see, we do not need them, at least with our current E2E tests.

> Manual re-run solved the issue and I believe that the failure could be related to GH actions.

Looking at the error I quoted above, I think it's an issue when trying to install `xdebug`:

```
The command '/bin/sh -c pecl install xdebug-2.9.8 [...] returned a non-zero code: 1
```

File: tests/e2e/env/docker-compose.yml (hunk 1 above)
Comment (htdat) [reply tests/e2e/env/docker-compose.yml]: Addressed here: https://github.com/Automattic/woocommerce-payments/pull/4577/commits/075050509db89848271367782d8b03c868c93468

**Comment by htdat** [comment]
Addressed feedback. Ready for another review. 

@achyuthajoy - since I will be off soon, please help me merge this PR if we get consensus about this change.

File: tests/e2e/env/docker-compose.yml (hunk 1 above)
Comment (achyuthajoy) [reply tests/e2e/env/docker-compose.yml]: Awesome. Thanks 👍

**Comment by achyuthajoy** [comment]
> Addressed feedback. Ready for another review.
> 
> @achyuthajoy - since I will be off soon, please help me merge this PR if we get consensus about this change.

Sure @htdat. I'll merge the PR when approved.

File: tests/e2e/env/docker-compose.yml [hunk 3]
```diff
@@ -2,8 +2,7 @@ version: '3'
 
 services:
   wordpress:
-    build: ./wordpress-xdebug
-    image: wcp_e2e_wordpress
+    image: wordpress:5.8.3-php7.4
```
Comment (deepakpathania) [inline tests/e2e/env/docker-compose.yml:5]: I am ok with the change as well, only concern is whether updating this image as we bump versions would become an additional task for us. Any way to get the `latest` we support here?

Comment (achyuthajoy) [reply tests/e2e/env/docker-compose.yml:5]: I don't think it's an issue @deepakpathania. We update the WP & WC versions before running tests.

Comment (htdat) [reply tests/e2e/env/docker-compose.yml:5]: > We update the WP & WC versions before running tests.

Yes, that's right. Also, if using `latest`, I am not sure if we can run the downgrade with this command: 
https://github.com/Automattic/woocommerce-payments/blob/5d19129845e132624bb08cf8f7bb325803f04516/tests/e2e/env/setup.sh#L152-L155

Based on the description here https://developer.wordpress.org/cli/commands/core/update/, it's for `new versions`.

**Comment by achyuthajoy** [comment]
Closing the PR based on the discussion to keep development & test environment in sync. With the suggested change we would be using a different docker image from what's present on development environment.

**Pull request closed without merging**